hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/event/WriteMetadataEvent.java

```diff
@@ -38,6 +38,7 @@ public class WriteMetadataEvent implements OperatorEvent {
 
   private List<WriteStatus> writeStatuses;
   private int taskID;
+  private long checkpointId;
```

Comment:
Upgrading existing Flink ingestion job with the current PR may cause compatibility issues, such as state restore failure

Reply:
yeah, it looks like there is no good way to make it compatible? If the recovered checkpoint id is -1, it still works for most of the cases.

Reply:
For `WriteMetaDataEvent` state, it uses `PojoSerializer` as state serializer which supports basic [state schema evolution](https://nightlies.apache.org/flink/flink-docs-release-2.0/docs/dev/datastream/fault-tolerance/serialization/schema_evolution/#pojo-types), including adding new fields, but the new added fields will 
 be default value when recovering from legacy state, i.e., always be -1 for checkpointId.

Reply:
We have an analysis again, if the job was recovered with some inflight instant events for the first time, these events would be recommitted into the table, the new job checkpoint id will start from -1.